lib/internal/streams/end-of-stream.js

```diff
@@ -35,7 +35,8 @@ function eos(stream, opts, callback) {
     if (!stream.writable) onfinish();
   };
 
-  var writableEnded = stream._writableState && stream._writableState.finished;
+  var writableEnded = stream.writableFinished ||
```

Comment:
The whole point of these new getters is to avoid using `{_readable,_writable}State` but in this case if the getter exists and returns `false` we use `_writableState` twice (first time in the getter itself and second time in the RHS of the "or" condition) so I'm not a fan of this change.

Reply:
don't you think V8 is able to figure that out?

Reply:
Perhaps, I don't know.

Reply:
Is there a desire to use `writableFinished` instead of `_writableState` in files such as this one, that are internal to streams, or should we only be doing that outside of the streams module itself (for example, in `lib/http.js`).  The answer is relevant to determining how much work remains to be done to close what is literally the oldest open issue we have: https://github.com/nodejs/node/issues/445.

Reply:
Oh, wait, I see there's the "Improves compat with streamlike objects." comment, so the motivation isn't just expanding usage of `writableFinished` for its own sake.

Reply:
Good point, I also missed that. It makes more sense now, approving.